src/assetmanagement.rs

```diff
@@ -98,3 +99,73 @@ where
         handle
     }
 }
+
+pub mod util {
+    use amethyst::{
+        assets::{AssetStorage, Loader},
+        ecs::{Entity, ReadExpect, Write, WriteStorage},
+        renderer::{
+            Material, MaterialDefaults, Mesh, MeshHandle, ObjFormat, PngFormat, Texture,
+            TextureMetadata,
+        },
+    };
+    use assetmanagement::AssetManager;
+
+    /// A collection of the required resources necessary to attach `Texture` and `MeshHandle` to an entity
+    pub type AssetStorages<'a> = (
+        ReadExpect<'a, Loader>,
+        Write<'a, AssetManager<Mesh>>,
+        WriteStorage<'a, MeshHandle>,
+        Write<'a, AssetStorage<Mesh>>,
+        Write<'a, AssetManager<Texture>>,
+        WriteStorage<'a, Material>,
+        Write<'a, AssetStorage<Texture>>,
+        ReadExpect<'a, MaterialDefaults>,
+    );
+
+    /// Attaches `Material` and `MeshHandle` of specified asset to the entity
+    pub fn insert_into_asset_storages(
+        entity: Entity,
+        asset_name: &str,
+        storages: &mut AssetStorages,
+    ) {
+        let (
+            ref loader,
+            ref mut mesh_manager,
+            ref mut mesh_handles,
+            ref mut mesh_storage,
+            ref mut tex_manager,
+            ref mut mat_storage,
+            ref mut tex_storage,
+            ref default_mat,
+        ) = storages;
+
+        let mesh = {
+            mesh_manager.get_asset_handle_or_load(
+                asset_name,
+                ObjFormat,
+                Default::default(),
+                mesh_storage,
+                &loader,
+            )
+        };
+
+        // load texture/material
+        let material = {
+            let handle = tex_manager.get_asset_handle_or_load(
+                asset_name,
+                PngFormat,
+                TextureMetadata::srgb(),
+                tex_storage,
```

Comment:
no it is not, because everything else would result in invalid borrows at runtime (double borrows.) but with the new assetmangers we are able to remove some of them like `loader, defaultmats`